### ISSUE TITLE
Move to typescript breaks hot-reload.

### ISSUE THREAD

**Issue by sergiop** [insin/nwb#531]
This issue is a **Support request**

Hi I moved my `nwb` based project to Typescript. Everything fine except for the automatic reload of the browser on save.

 I created the following `webpack` custom config in `nwb.config.js` and the following dependencies.

What I'am missing?
```
yarn add @types/react @types/react-dom awesome-typescript-loader typescript -D
```
```
  webpack: {
    config(config) {
      config.entry = {
        demo: ['./demo/src/index.tsx'],
      }
      config.resolve.extensions = []
      config.resolve.extensions.push('.ts', '.tsx', '.jsx', '.js')
      config.devtool = 'source-map'
      config.module.rules.push({
        test: /\.tsx?$/,
        loaders: ['awesome-typescript-loader'],
      })

      return config
    },
  },
```

**Comment by ironsq** [comment]
I also ran into this issue and it seems that it's the entrypoint configuration that breaks the hot reloading

`config.entry = {
                demo: ["./demo/src/index.tsx"]
            };`

Remove the above and you will re-enable it. However, the webpack compilation will fail since it the default entrypoint will be missing. Adding index.js to the src folder will solve this seemingly, hot reloading will start as per usual but will throw a warning whenever you modify index.js and will ignore any changes to any other .tsx file in your project.

I created a simple webpack.config.js in a nwb created react-app, using nwb dependencies from 0.23.0 only adding webpack-cli version 3.2.1 (any later version will not allow webpack to finish compiling my project) and that will enable Typescript support with hot reloading and a custom entrypoint.

In conclusion, it seems that it's not the addition of ts-loader or awesome-typescript-loader that breaks hot reloading, it's defining the entrypoint in the webpack section. It does not matter if you use the escape hatch (extra) or provide your own config(config).

**Comment by wizzard0** [comment]
for me, as a workaround (I prefer to build components together with the main project, not when publishing) keeping `tsc --watch` alongside nwb and using `.js` as entry point works fine (including HMR)

**Comment by shivangsanghi** [comment]
@sergiop When i build the project for the UMD, it works fine. But when i try to use that UMD build, the global variable is set to undefined. 

It sets correctly without typescript configurations.

**Comment by Nikodermus** [comment]
For TS, my module had only named exports, so I added the default for all the named exports, which worked with the UMD build

```javascript
// Required for drop a tag configuration
export default {
    CAMERA_FOCUS_DURATION,
    CAMERA_PITCH,
    CAMERA_RESET_DURATION,
    CAMERA_ZOOM_DURATION,
    MAP_CONTAINER_CLASS,
    ROW_ZOOM_OFFSET,
    SEATING_ID_PATTERN,
    VenueMapComponent,
    VenueMapErrorEvent,
    VenueMapEvent,
    VenueMapFocusOptions,
    VenueMapInputEvent,
    VenueMapOptions,
    VenueMapRowInputEvent,
    VenueMapSectionInputEvent
};
```
